landoapi/landing_worker.py

```diff
@@ -31,17 +32,25 @@
 
 
 @contextmanager
-def job_processing(worker):
-    """Mutex-like context manager that ensures workers shut down gracefully.
+def job_processing(worker, job, db):
+    """Mutex-like context manager that manages job processing miscellany.
+
+    This context manager facilitates graceful worker shutdown, tracks the duration of
+    the current job, and commits changes to the DB at the very end.
 
     Args:
         worker (LandingWorker): the landing worker that is processing jobs
+        job (LandingJob): the job currently being processed
+        db (SQLAlchemy): active database session
     """
     worker.job_processing = True
+    start_time = datetime.now()
     try:
         yield
     finally:
         worker.job_processing = False
+        job.duration_seconds = (datetime.now() - start_time).seconds
```

Comment:
This only records the amount of time for the most recent landing attempt. I think this is fine for our purposes but I thought I'd call that out.